radixdlt-core/radixdlt/src/main/java/com/radixdlt/tree/PMT.java

```diff
@@ -101,16 +95,30 @@
   // verify_proof
 
   public PMT(PMTStorage db) {
-    this(db, new SHA256(), new RLPSerializer(), Duration.of(10, ChronoUnit.MINUTES));
+    this(db, (byte[]) null, new SHA256(), new RLPSerializer());
+  }
+
+  public PMT(PMTStorage db, byte[] currentRoot) {
+    this(db, currentRoot, new SHA256(), new RLPSerializer());
+  }
+
+  public PMT(PMTStorage db, HashFunction hashFunction, PMTNodeSerializer pmtNodeSerializer) {
+    this(db, (byte[]) null, hashFunction, pmtNodeSerializer);
+  }
+
+  public PMT(
+      PMTStorage db, byte[] root, HashFunction hashFunction, PMTNodeSerializer pmtNodeSerializer) {
+    this(
+        db,
+        root == null ? null : pmtNodeSerializer.deserialize(root),
+        hashFunction,
+        pmtNodeSerializer);
   }
 
   public PMT(
-      PMTStorage db,
-      HashFunction hashFunction,
-      PMTNodeSerializer pmtNodeSerializer,
-      Duration cacheExpiryAfter) {
+      PMTStorage db, PMTNode root, HashFunction hashFunction, PMTNodeSerializer pmtNodeSerializer) {
     this.db = db;
-    this.cache = new PMTCache(CACHE_MAXIMUM_SIZE, cacheExpiryAfter, this::nodeLoader);
+    this.root = root;
     this.hashFunction = hashFunction;
     this.pmtNodeSerializer = pmtNodeSerializer;
     this.emptyTreeHash = hashFunction.hash(pmtNodeSerializer.emptyTree());
```

Comment:
we could save a single hash op and remove the emptyTreeHash property if we extend HashFunction interface with `byte[] emptyHash();` (we could then rename it to something like `PMTHasher`)

Reply:
the tree hash depends on both the serialiser and hash function